[CI] RolloverIT testRollover failing

Suspect this is a general problem related to #83576.

**Build scan:**
https://gradle-enterprise.elastic.co/s/zxsmvjenj7xro/tests/:server:internalClusterTest/org.elasticsearch.action.admin.indices.rollover.RolloverIT/testRollover

**Reproduction line:**
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.action.admin.indices.rollover.RolloverIT.testRollover" -Dtests.seed=58E68D6FC10A7415 -Dtests.locale=fr-CA -Dtests.timezone=America/Lower_Princes -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.action.admin.indices.rollover.RolloverIT&tests.test=testRollover

**Failure excerpt:**
```
com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=1168, name=elasticsearch[node_t0][scheduler][T#1], state=RUNNABLE, group=TGRP-RolloverIT]

  at __randomizedtesting.SeedInfo.seed([58E68D6FC10A7415:D3FB80172FA618EA]:0)

  Caused by: org.elasticsearch.common.util.concurrent.EsRejectedExecutionException: rejected execution of detected failure of {node_t1}{EahwztF4RJCdtcO0EytfvQ}{LuW7WWgCS4eGeQA7k6TZpw}{127.0.0.1}{127.0.0.1:33989}{cdfhilrstw} on EsThreadPoolExecutor[name = node_t0/cluster_coordination, org.elasticsearch.common.util.concurrent.EsThreadPoolExecutor@4939db0f[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 140]] (shutdown)

    at __randomizedtesting.SeedInfo.seed([58E68D6FC10A7415]:0)
    at org.elasticsearch.common.util.concurrent.EsRejectedExecutionHandler.newRejectedException(EsRejectedExecutionHandler.java:36)
    at org.elasticsearch.common.util.concurrent.EsAbortPolicy.rejectedExecution(EsAbortPolicy.java:34)
    at java.util.concurrent.ThreadPoolExecutor.reject(ThreadPoolExecutor.java:833)
    at java.util.concurrent.ThreadPoolExecutor.execute(ThreadPoolExecutor.java:1365)
    at org.elasticsearch.common.util.concurrent.EsThreadPoolExecutor.execute(EsThreadPoolExecutor.java:66)
    at org.elasticsearch.cluster.coordination.FollowersChecker$FollowerChecker.failNode(FollowersChecker.java:361)
    at org.elasticsearch.cluster.coordination.FollowersChecker$FollowerChecker$1.handleException(FollowersChecker.java:354)
    at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:665)
    at org.elasticsearch.cluster.coordination.FollowersChecker$FollowerChecker.handleWakeUp(FollowersChecker.java:301)
    at org.elasticsearch.cluster.coordination.FollowersChecker$FollowerChecker$3.run(FollowersChecker.java:387)
    at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:717)
    at org.elasticsearch.threadpool.ThreadPool$1.run(ThreadPool.java:423)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
    at java.util.concurrent.FutureTask.run(FutureTask.java:264)
    at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
    at java.lang.Thread.run(Thread.java:833)

```

--- a/server/src/main/java/org/elasticsearch/cluster/coordination/FollowersChecker.java
+++ b/server/src/main/java/org/elasticsearch/cluster/coordination/FollowersChecker.java
@@ -19,6 +19,7 @@ import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.util.concurrent.AbstractRunnable;
+import org.elasticsearch.common.util.concurrent.EsRejectedExecutionException;
 import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.monitor.NodeHealthService;
 import org.elasticsearch.monitor.StatusInfo;
@@ -358,9 +359,20 @@ public class FollowersChecker {
         }
 
         void failNode(String reason) {
-            transportService.getThreadPool().executor(Names.CLUSTER_COORDINATION).execute(new Runnable() {
+            transportService.getThreadPool().executor(Names.CLUSTER_COORDINATION).execute(new AbstractRunnable() {
+
                 @Override
-                public void run() {
+                public void onRejection(Exception e) {
+                    logger.debug(new ParameterizedMessage("rejected task to fail node [{}] with reason [{}]", discoveryNode, reason), e);
+                    if (e instanceof EsRejectedExecutionException esRejectedExecutionException) {
+                        assert esRejectedExecutionException.isExecutorShutdown();
+                    } else {
+                        assert false : e;
+                    }
+                }
+
+                @Override
+                protected void doRun() {
                     synchronized (mutex) {
                         if (running() == false) {
                             logger.trace("{} no longer running, not marking faulty", FollowerChecker.this);
@@ -371,6 +383,15 @@ public class FollowersChecker {
                         followerCheckers.remove(discoveryNode);
                     }
                     onNodeFailure.accept(discoveryNode, reason);
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    assert false : e;
+                    logger.error(
+                        new ParameterizedMessage("unexpected failure when failing node [{}] with reason [{}]", discoveryNode, reason),
+                        e
+                    );
                 }
 
                 @Override

